src/usePopper.js

```diff
@@ -29,28 +71,58 @@ export default function usePopper(
   {
     enabled = true,
     placement = 'bottom',
-    positionFixed = false,
+    strategy = 'absolute',
     eventsEnabled = true,
-    modifiers = {},
+    ...options
   } = {},
 ) {
   const popperInstanceRef = useRef();
 
-  const hasArrow = !!(modifiers.arrow && modifiers.arrow.element);
-
   const scheduleUpdate = useCallback(() => {
     if (popperInstanceRef.current) {
-      popperInstanceRef.current.scheduleUpdate();
+      popperInstanceRef.current.update();
     }
   }, []);
 
-  const [state, setState] = useState({
-    placement,
-    scheduleUpdate,
-    outOfBoundaries: false,
-    styles: initialPopperStyles,
-    arrowStyles: initialArrowStyles,
-  });
+  const [state, setState] = useSafeState(
+    useState({
+      placement,
+      scheduleUpdate,
+      outOfBoundaries: false,
+      styles: initialPopperStyles,
+      arrowStyles: initialArrowStyles,
+    }),
+  );
+
+  const updateModifier = useMemo(
+    () => ({
+      name: 'updateStateModifier',
+      enabled: true,
+      phase: 'afterWrite',
+      requires: ['computeStyles'],
+      fn: data => {
+        setState({
+          scheduleUpdate,
+          outOfBoundries: data.state.modifiersData.hide?.isReferenceHidden,
+          placement: data.state.placement,
+          styles: { ...data.state.styles?.popper },
+          arrowStyles: { ...data.state.styles?.arrow },
+          state: data.state,
+        });
+      },
+    }),
+    [scheduleUpdate, setState],
+  );
+  const modifiers = toModifierArray(options.modifiers);
+
+  let eventsModifier = modifiers.find(m => m.name === 'eventListeners');
+  if (!eventsModifier && eventsEnabled) {
+    eventsModifier = {
+      name: 'eventListeners',
+      enabled: true,
+    };
+    modifiers.push(eventsModifier);
```

Comment:
this is slightly odd in that it may mutate `options.modifiers` (if it was an array), which seems odd.

instead of `toModifierMap`, &c., you might be better off adding some sort of `mergeModifiers` abstraction that's guaranteed to never mutate the input